app/parsers/bulkrax/application_parser.rb

```diff
@@ -37,15 +38,94 @@ def records(opts = {})
       raise 'must be defined'
     end
 
-    def import_fields
-      raise 'must be defined'
+    def import_file_path
+      @import_file_path ||= self.parser_fields['import_file_path']
+    end
+
+    def create_collections
+      raise 'must be defined' if importer?
+    end
+
+    def create_works
+      raise 'must be defined' if importer?
+    end
+
+    # Optional, only used by certain parsers
+    # Other parsers should override with a custom or empty method
+    # Will be skipped unless the record is a Hash
+    def create_parent_child_relationships
+      parents = setup_parents
+      return if parents.blank?
+
+      parents.each do | key, value |
+        parent = entry_class.where(
+          identifier: key,
+          importerexporter_id: importerexporter.id,
+          importerexporter_type: 'Bulkrax::Importer',
+        ).first
+
+        # not finding the entries here indicates that the given identifiers are incorrect
+        # in that case we should log that
+        children = []
+        children = value.map do | child |
+          entry_class.where(
+            identifier: child,
+            importerexporter_id: importerexporter.id,
+            importerexporter_type: 'Bulkrax::Importer',
+          ).first
+        end.compact.uniq
+
+        if parent.present? && (children.length != value.length)
+          # Increment the failures for the number we couldn't find
+          # Because all of our entries have been created by now, if we can't find them, the data is wrong
+          Rails.logger.error("Expected #{value.length} children for parent entry #{parent.id}, found #{children.length}")
+          return if children.length == 0
+          Rails.logger.warn("Adding #{children.length} children to parent entry #{parent.id} (expected #{value.length})")
+        end
+        parent_id = parent.id
+        child_entry_ids = children.map { |c| c.id }
+        ChildRelationshipsJob.perform_later(parent_id, child_entry_ids, current_importer_run.id)
+      end
+    end
+
+    def setup_parents
```

Comment:
OO wise, I think it might make sense to do `def parents` and `@parents` instead of setup_parents and not persist them.